sunpy/map/sources/sdo.py

```diff
@@ -110,6 +110,17 @@ def __init__(self, data, header, **kwargs):
         self.meta['detector'] = self.meta.get('detector', "HMI")
         self._nickname = self.detector
 
+    @property
+    def unit(self):
```

Comment:
What's all this for?

Reply:
HMI uses a non-standard unit, so this is allowing it through.

Reply:
I think it would be better to fix this in the map source, and emit a debug log at the same time, which is what is currently done elsewhere to fix units.

Reply:
This is the map source? The issue is I don't think we can transform the header to a FITS standard unit, because the array is in Gauss and Gauss isn't an allowed unit.

Reply:
Hmm, https://fits.gsfc.nasa.gov/standard40/fits_standard40aa-le.pdf Table 4 (page 9) says it's allowed.

Reply:
So it looks like 'Gauss' just needs correcting to 'G'.